include/zephyr/linker/common-rom/common-rom-misc.ld

```diff
@@ -17,6 +17,13 @@
 	} GROUP_LINK_IN(ROMABLE_REGION)
 #endif /* CONFIG_EMUL */
 
+	SECTION_DATA_PROLOGUE(symbol_to_keep,,)
+	{
+		__symbol_to_keep_start = .;
+		KEEP(*(SORT(.symbol_to_keep*)));
+		__symbol_to_keep_end = .;
+	} GROUP_ROM_LINK_IN(RAMABLE_REGION, ROMABLE_REGION)
```

Comment:
This is still in its own section internal to the image memory, so it's occupying flash or sram at runtime.   It shouldn't be hard to put this e.g. at the end of the link and pull it out with objcopy, etc...

Not worth -1, but I personally find the waste irksome.

Reply:
@tejlmand ^^

Reply:
It can always be improved and won't change macro behavior. Current use case is for test applications so it does not matter that much.

Reply:
I think @andyross has a very valid point.

I won't block this PR for it, but i'm concerned that if such improvement is left for the future it will simply be forgotten or down prioritized, so if you have the time now @nordic-krch then I believe a good time to do it would be now 😄